src/plugins/proposal.ts

```diff
@@ -769,4 +771,63 @@ export abstract class Proposal<TProposalState extends IProposalState> extends En
 
     return toIOperationObservable(observable)
   }
+
+  public redeemerContract() {
+    for (const version of REDEEMER_CONTRACT_VERSIONS) {
+      try {
+        const contractInfo = this.context.getContractInfoByName('Redeemer', version)
+        return this.context.getContract(contractInfo.address)
+      } catch (err) {
+        if (!err.message.match(/no contract/i)) {
+          // if the contract cannot be found, try the next one
+          throw err
+        }
+      }
+    }
+    throw Error(
+      `No Redeemer contract could be found (search for versions ${REDEEMER_CONTRACT_VERSIONS})`
+    )
+  }
+
+  public redeemRewards(beneficiary?: Address): Operation<boolean> {
+    const mapReceipt = (receipt: ITransactionReceipt) => true
+
+    const createTransaction = async (): Promise<ITransaction> => {
+      if (!beneficiary) {
+        beneficiary = NULL_ADDRESS
+      }
+
+      const state = await this.fetchState()
+      let pluginAddress
+      if (state.name === 'ContributionReward') {
+        const pluginState = await state.plugin.entity.fetchState()
+        pluginAddress = pluginState.address
+      } else {
```

Comment:
why this condition is needed ?

Reply:
The old functions are not exactly the same - they are similar but not the same. There is a different in the way `pluginAddress` is fetched between the functions and we need to distinguish between the ways. Without this condition some of the tests fail.

Reply:
there is a need to understand why this behave differently .
does `pluginAddress` get different value for different cases ? why ?

Reply:
Yes it gets a different value when `state.name` is `ContributionReward` or different.